client/my-sites/github-deployments/connection-wizard-button.tsx

```diff
@@ -0,0 +1,87 @@
+import config from '@automattic/calypso-config';
+import page from '@automattic/calypso-router';
+import { Button } from '@automattic/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { useState } from 'react';
+import SocialLogo from 'calypso/components/social-logo';
+import { useSelector } from 'calypso/state';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import { useGithubAccountsQuery } from './use-github-accounts-query';
+
+const AUTHORIZE_URL = addQueryArgs( 'https://github.com/login/oauth/authorize', {
+	client_id: config( 'github_oauth_client_id' ),
+} );
+
+const POPUP_ID = 'github-oauth-authorize';
+
+const authorizeGitHub = () => {
+	return new Promise< void >( ( resolve, reject ) => {
+		const popup = window.open( AUTHORIZE_URL, POPUP_ID, 'height=600,width=700' );
+
+		if ( ! popup ) {
+			reject();
+		}
+
+		const interval = setInterval( (): void => {
+			if ( popup?.closed ) {
+				resolve();
+				clearInterval( interval );
+			}
+		}, 500 );
+	} );
+};
+
+export const ConnectionWizardButton = () => {
+	const { __ } = useI18n();
+
+	const {
+		data: accounts,
+		isLoading,
+		isRefetching,
+		refetch,
+	} = useGithubAccountsQuery( {
+		retry: false,
+	} );
+
+	const [ isAuthorizing, setIsAuthorizing ] = useState( false );
+	const siteSlug = useSelector( getSelectedSiteSlug );
+
+	const hasAuthorized = !! accounts;
+
+	const startAuthorization = () => {
+		setIsAuthorizing( true );
+
+		authorizeGitHub().then( () => {
+			setIsAuthorizing( false );
+			refetch();
+		} );
```

Comment:
I see `authorizeGitHub` handling the case where the popup is blocked. I'm thinking maybe we should move the `setIsAuthorizing()` into a `finally` block so that the button doesn't stay disabled if the popup is blocked.

```suggestion
		} ).finally( () => {
			setIsAuthorizing( false );
		} );
```